source/common/router/router.cc

```diff
@@ -209,11 +209,25 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
     return Http::FilterHeadersStatus::StopIteration;
   }
 
-  // Determine if there is a redirect for the request.
-  if (route_->redirectEntry()) {
-    config_.stats_.rq_redirect_.inc();
-    Http::Utility::sendRedirect(*callbacks_, route_->redirectEntry()->newPath(headers),
-                                route_->redirectEntry()->redirectResponseCode());
+  // Determine if there is a direct response for the request.
+  if (route_->directResponseEntry()) {
+    auto response_code = route_->directResponseEntry()->responseCode();
+    if (response_code >= Http::Code::MultipleChoices && response_code < Http::Code::BadRequest) {
+      config_.stats_.rq_redirect_.inc();
+      Http::Utility::sendRedirect(*callbacks_, route_->directResponseEntry()->newPath(headers),
+                                  response_code);
+      return Http::FilterHeadersStatus::StopIteration;
+    }
+    Http::Utility::sendLocalReply(
```

Comment:
Take a look at callers of sendLocalReply - I think you can simplify the arguments a bit (and if I'm incorrect please educate me! :-P)

Reply:
Oh, I see, this method is in the same class that provides a friendly wrapper around sendLocalReply. I'll switch to that.